### ISSUE TITLE
RTE inserts <style> tag whose classes override global styles

### ISSUE THREAD

**Issue by sandorvasas** [syncfusion/ej2-react-ui-components#17]
Is there a way to prefix those inserted classes, or add a hash to them so they would only apply to contents inside the RTE?

**Comment by vinoth-Kumar-S** [comment]
Hi Sandorvasa,

Good day to you.

You can use the **cssClass** property for setting the custom class name for the RichTextEditor component. This class name appends to root element of the RichTextEditor.

**API link:** https://ej2.syncfusion.com/documentation/api/rich-text-editor#cssclass

In below sample, we have rendered the unordered list into RTE component and outside the component and used the same class name on ‘**e-paragraph**’. Also, we have added the **cssClass** property for the root element of RTE component as ‘**e-rteClass**’ and applied the color inside the RTE component alone.

**Sample Link:** https://stackblitz.com/edit/6tflca-1yycac?file=index.html

Could you please check the above sample and If the above solution does not fulfill your requirement, get back to us with your code and replicate your problem on the shared sample above. So, we'll check and update the timely solution as soon as possible.

Regards,
Vinoth Kumar S

**Comment by sandorvasas** [comment]
I checked the code example, but it doesn't cover my case. What you are doing is applying CSS classes to parts of the RTE content. 
What my problem is, is that a <style></style> tag appeared in the contents of the RTE (by itself, by user input, they inserted a link <a> or something, I don't know), and that style tag is as follows: 

```css
<style>
a:link { color: blue; text-decoration: underline }
</style>
```

It's global styling that overrides our app UI's style in the Navbar, and other places.
The style being inserted into the RTE should be prefixed with something specific to the RTE, like `.e-RTE-class a`

**Comment by ChristoIssac** [comment]
Hi Sandorvasa,

Good day to you.

We have analyzed your query. Already, we have used specific Rich Text Editor class name for all our Rich Text Editor content like e-rte-content, e-rte-table, e-rte-image and e-rte-anchor. Also, we have applied styles based on hierarchical class name. For example, our anchor tag color is applied using the class name ‘.e-richtexteditor .e-rte-content .e-content a’. We didn’t use any styles which you shared in RTE.

Can you please share below information,
 
•	Where you are facing the problem and which Rich Text Editor content class is overridden on global styles?
•	Can you please reproduce your issue on the sample we have provided? or else please share your code or screenshot

Thanks,
Christo

**Comment by sandorvasas** [comment]
The following CSS appeared somewhere around an <ul> tag in the editor content:
```css

		@page { size: 21.59cm 27.94cm; margin: 2cm }
		p { direction: ltr; text-align: justify; orphans: 2; widows: 2; background: transparent }
		p.western { font-size: 12pt; so-language: en-AU; font-style: italic }
		p.cjk { font-size: 12pt; so-language: en-US; font-style: italic }
		a:link { color: #0000ff; text-decoration: underline }
```
I don't know how to reproduce.

**Comment by PrinceOliver** [comment]
Hi @sandorvasas ,

As already stated, we have defined all styles on Rich Text Editor component element based on hierarchical class name structure. So, the styles will not affect the global styles. We have not used your shared CSS code anywhere within the editor content. We were not able to replicate your issue at our end. We have already added prefix class name **‘e-rte-’** to most of the classes on Editor content. We suspect that you might have installed any unwanted plugins in your machine. Kindly ensure this in your end and share adequate data or code with us to replicate the problem in our end. This will help us proceed further.

Regards,
Prince
